templates/repo/commits_list_small.tmpl

```diff
@@ -31,28 +31,48 @@
 					{{$class = (printf "%s%s" $class " isWarning")}}
 				{{end}}
 			{{end}}
-			{{if $.Issue.PullRequest.BaseRepo.Name}}
-				<a href="{{AppSubUrl}}/{{$.Issue.PullRequest.BaseRepo.OwnerName}}/{{$.Issue.PullRequest.BaseRepo.Name}}/commit/{{.ID}}" rel="nofollow" class="{{$class}}">
+			{{if $.comment.Issue.PullRequest.BaseRepo.Name}}
+				<a href="{{AppSubUrl}}/{{$.comment.Issue.PullRequest.BaseRepo.OwnerName}}/{{$.comment.Issue.PullRequest.BaseRepo.Name}}/commit/{{.ID}}" rel="nofollow" class="{{$class}}">
 			{{else}}
 				<span class="{{$class}}">
 			{{end}}
-					<span class="shortsha">{{ShortSha .ID.String}}</span>
-			{{if $.Issue.PullRequest.BaseRepo.Name}}
+			<span class="shortsha">{{ShortSha .ID.String}}</span>
+			{{if .Signature}}
+				<div class="ui detail icon button">
+					{{if .Verification.Verified}}
+						<div title="{{if eq .Verification.TrustStatus "trusted"}}{{else if eq .Verification.TrustStatus "untrusted"}}{{$.root.i18n.Tr "repo.commits.signed_by_untrusted_user"}}: {{else}}{{$.root.i18n.Tr "repo.commits.signed_by_untrusted_user_unmatched"}}: {{end}}{{.Verification.Reason}}">
+						{{if ne .Verification.SigningUser.ID 0}}
+							<i class="lock icon"></i>
+							<img class="ui signature avatar image" src="{{.Verification.SigningUser.RelAvatarLink}}" />
+						{{else}}
+							<i title="{{.Verification.Reason}}" class="icons">
+								<i class="lock icon"></i>
+								<i class="tiny inverted cog icon centerlock"></i>
+							</i>
+							<img class="ui signature avatar image" src="{{AvatarLink .Verification.SigningEmail}}" />
+						{{end}}
+						</div>
+					{{else}}
+						<i title="{{$.root.i18n.Tr .Verification.Reason}}" class="unlock icon"></i>
+					{{end}}
+				</div>
```

Comment:
Would it make sense to extract this badge to its own template file? This is the third occurence of it. [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) 😉

Reply:
Nested templates within nested template within nested template within nested template :>

Will try.

Reply:
Would this be enough?

Reply:
Looking good at first glance. Will test later.